src/OrchardCore/OrchardCore/Modules/PoweredByMiddleware.cs

```diff
@@ -1,48 +1,34 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
 
 namespace OrchardCore.Modules
 {
     /// <summary>
-    /// Adds the X-Powered-By header with values OrchardCore.
+    /// Represents a middlware that Adds X-Powered-By HTTP header with the value <c>Orchard Core</c>.
     /// </summary>
     public class PoweredByMiddleware
     {
+        private readonly PoweredByOptions _options;
         private readonly RequestDelegate _next;
-        private readonly IPoweredByMiddlewareOptions _options;
 
-        public PoweredByMiddleware(RequestDelegate next, IPoweredByMiddlewareOptions options)
+        /// <summary>
+        /// Creates an instance of <see cref="PoweredByMiddleware"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="IOptions{TOptions}"/>.</param>
+        /// <param name="next">The <see cref="RequestDelegate"/>.</param>
+        public PoweredByMiddleware(IOptions<PoweredByOptions> options, RequestDelegate next)
         {
+            _options = options.Value;
             _next = next;
-            _options = options;
         }
 
+        /// <inheritdoc/>
         public Task Invoke(HttpContext httpContext)
         {
-            if (_options.Enabled)
-            {
-                httpContext.Response.Headers[_options.HeaderName] = _options.HeaderValue;
-            }
+            httpContext.Response.Headers.XPoweredBy = _options.Value;
 
             return _next.Invoke(httpContext);
         }
     }
-
-    public interface IPoweredByMiddlewareOptions
-    {
-        bool Enabled { get; set; }
-        string HeaderName { get; }
-        string HeaderValue { get; set; }
-    }
-
-    internal class PoweredByMiddlewareOptions : IPoweredByMiddlewareOptions
-    {
-        private const string PoweredByHeaderName = "X-Powered-By";
-        private const string PoweredByHeaderValue = "OrchardCore";
-
-        public string HeaderName => PoweredByHeaderName;
-        public string HeaderValue { get; set; } = PoweredByHeaderValue;
-
-        public bool Enabled { get; set; } = true;
```

Comment:
Can we still disable it?

Reply:
The idea of any middleware you can enable it when you inject. I'm not sure if there's a value to diable it!!

Reply:
Don't fogot this an indicator that the website or web app is powered by OC, also it would be nice to keep Orchard Core value exist even the user supply a custom value

Reply:
Not a problem if we don't want anymore to be able to disable it, otherwise, because we always add this middleware in `UseOrchardCore()`, we would still need that the middleware inject and use dedicated options.

Just saw that currently the options class is an `IPoweredByMiddlewareOptions` registered as a singleton, if we keep the options would be better to use the regular `IOptions` and `IConfiguration<>` patterns, currently it is enabled by default, and you can disable it by calling in your app startup `app.UsePoweredByOrchardCore(false)` which set `Enabled` to false.

Reply:
> if we keep the options would be better to use the regular IOptions and IConfiguration<> patterns,

I removed the singleton registration and used Options pattern instead

Reply:
Yes, you use a `PoweredByOptions` but not the config options pattern, you create a new options with a default value and pass it as a parameter to `app.UseMiddleware<>(options)`, so the middleware will not inject options from the DI, so we can't collaborate/change options from the app startup (here we are at the app level), this to change the value or disable it if still allowed.

Reply:
Ya I didn't use config options, but I'm using pattern instead coz nothing need to read from config

>  so we can't collaborate/change options from the app startup (here we are at the app level), this to change the value or disable it if still allowed.

I didn't know how this make a change, while I removed the `Enabled` property. I think the intend here is to make the header present all the time

Reply:
Yes indeed if we don't allow to disable it anymore, that's why I mentioned if we still allow to disable it (related to the Seb question), but I like your answer ;)

So, if nothing is configurable why passing `Options.Create(options)` to the middleware, even more, not saying we should do it, we could just use a const value defined in the middleware itself.

This remember me the discussion for CultureScope where I suggested to check if it switches the culture in place of just checking the value we pass, and you answered that anyway they are equal ;)

Reply:
> Yes indeed if we don't allow to disable it anymore, that's why I mentioned if we still allow to disable it (related to the Seb question), but I like your answer ;)

I see, may be I miss understood ;)

> So, if nothing is configurable why passing Options.Create(options) to the middleware, even more, not saying we should do it, we could just use a const value defined in the middleware itself.

IMHO the value is the only thing configurable here, where we can allow to add a new powered-by header. Again I think adding OrchardCore in both `PoweredByOrchardCore()` & `PoweredBy()`